pkg/config/config.go

```diff
@@ -586,6 +586,18 @@ func (c *Config) Validate() error {
 	return nil
 }
 
+func (c *EngineConfig) findRuntime() string {
+	// Search for crun first followed by runc and kata
+	for _, name := range []string{"crun", "runc", "kata"} {
```

Comment:
Hard coding the keys seems risky.  Shouldn't we iterate over the keys in the runtimes map (`c.OCIRuntimes) instead?

If the default runtime is set, we should only look for this specific one.

Reply:
The problem with that is we also want runc preferred over kata.  Remember this is only in effect if the user or distribution did not select an oci runtime in a containers.conf.

Reply:
I think these are two separate things.  One is the config, the other is the hierarchy.

A user may not set the runtime but set the runtimes map.  It may be worth to document the precedence of crun > runc > kata in the man page?

Reply:
Ah, it's already documented :) Thanks!